### PR TITLE
[AMD-ZENDNN] Environment variable to enable AMD-ZENDNN plugin

### PR DESCRIPTION
**Pull request by aadwived**
This pull request introduces the environment variable to enable AMD-ZENDNN Plugin
All code changes are under AMD_ZENDNN flag which is enabled for linux builds only.

Authors:
    Aakar Dwivedi ( aakar.dwivedi@amd.com )
    Chandra Kumar Ramasamy ( chandrakumar.ramasamy@amd.com )
    Savan Anadani ( savan.anadani@amd.com )

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>

### CODE REVIEW

File: tensorflow/core/util/port.cc [hunk 1]
```diff
@@ -116,4 +116,31 @@ bool IsMklEnabled() {
   return oneDNN_enabled;
 #endif  // ENABLE_MKL
 }
+
+bool IsZenDnnEnabled() {
+  static absl::once_flag once;
+#ifndef AMD_ZENDNN
+  return false;
+#else
+  static bool ZenDNN_enabled = false;
```
Comment (penpornk) [inline tensorflow/core/util/port.cc:125]: Once we get all the integration PRs in and things work fine, we may consider setting the default value to `true` if the plug-in is present. As of now I think it's good to start with false. :)

**Review by penpornk — APPROVED** [review]
Thank you very much for the PR!

File: tensorflow/core/util/port.cc (hunk 1 above)
Comment (aadwived) [reply tensorflow/core/util/port.cc:125]: Yes, that is a good idea. Thank you for the quick approval.

**Review by penpornk — CHANGES_REQUESTED** [review]

File: tensorflow/core/util/port.cc [hunk 2]
```diff
@@ -116,4 +116,31 @@ bool IsMklEnabled() {
   return oneDNN_enabled;
 #endif  // ENABLE_MKL
 }
+
+bool IsZenDnnEnabled() {
+  static absl::once_flag once;
```
[the remark this answers is not in the document]
Comment (aadwived) [reply tensorflow/core/util/port.cc]: Resolved. Thank you for the suggestion.

**Review by penpornk — APPROVED** [review]
Thank you for the quick change!

**Review by penpornk — APPROVED** [review]

**Pull request merged**